mlflow/pyfunc/__init__.py

```diff
@@ -1096,6 +1144,38 @@ def _predict_row_batch(predict_fn, args):
             StringType: np.str_,
         }
 
+        def _convert_array_values(values, spark_type):
+            array_dim = 1
+            elem_type = spark_type.elementType
+            if type(elem_type) == ArrayType:
+                elem_type = elem_type.elementType
+                if type(elem_type) == ArrayType:
```

Comment:
Can we use `isinstance`?

Reply:
```suggestion
            if type(elem_type) == ArrayType and type(elem_type.elementType) == ArrayType:
```

Reply:
The nested elem_type `elem_type = elem_type.elementType` is used in following code.